backend/app/api/endpoints/internal/rag_content.py

```
@@ -0,0 +1,90 @@
+    if token_info is None or token_info.attachment_id != attachment_id:
+        raise HTTPException(
+            status_code=status.HTTP_401_UNAUTHORIZED,
+            detail="Invalid RAG download token",
+        )
+
+
+@router.get("/{attachment_id}")
+async def stream_rag_attachment_content(
+    attachment_id: int,
+    _: None = Depends(_verify_rag_download_authorization),
+    db: Session = Depends(get_db),
```

Ruff (0.15.9)

[warning] Do not perform function call `Depends` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)